cartridge/feedback.lua

```diff
@@ -1,18 +1,68 @@
+local fio = require('fio')
 local log   = require('log')
 local json  = require('json')
 local fiber = require('fiber')
 local http  = require('http.client')
 
 local PREFIX = "feedback_daemon"
 
+local function parse_lua_file(name)
+    local state = {}
+    local eval = loadfile(name, 't', state)
+    pcall(eval)
+
+    return state
+end
+
+local function parse_manifest(filename)
+    local manifest = parse_lua_file(filename)
+    local deps = {}
+
+    for module, versions in pairs(manifest.dependencies or {}) do
+        for version, _ in pairs(versions) do
+            if deps[module] ~= nil then
+                break -- Decide how to deal with duplicates
+            end
+            deps[module] = version
+        end
+    end
+
+    return deps
+end
+
+local parsed_manifest = false
+
+local function fill_in_rocks(feedback)
+    feedback.rocks = feedback.rocks or {}
+
+    -- tarantool 1.10 doesn't support package.searchroot
+    -- so we are basically guessing by picking directory
+    -- of the init script. It is a default scenario
+    local app_dir = package.searchroot and package.searchroot() or fio.abspath(fio.dirname(arg[0]))
+    local manifest_path = fio.pathjoin(app_dir, '.rocks/share/tarantool/rocks/manifest')
+
+    if not parsed_manifest then
```

Comment:
This code isn't friendly to hot-reload. According to recent discussions in #842  I suggest to prepare for it.

Reply:
Add file modification time based caching for manifest. Also it will be invalidated if manifest when removed

Reply:
Decided not to deal with caching since feedback daemon runs only once in hour.